language/coalescer.go

```diff
@@ -21,9 +21,14 @@ func (c *coalescer) Coalesce(_ context.Context, ls []*indexer.LayerArtifacts) (*
 		Packages:     map[string]*claircore.Package{},
 		Repositories: map[string]*claircore.Repository{},
 	}
-
-	for _, l := range ls {
-		// If we didn't find at least one npm repo in this layer
+	// Similar to ir.Packages, except instead of mapping
+	// id -> package, it maps packageDB -> package.
+	// For langauge packages, it is possible the
+	// packageDB is overwritten.
```

Comment:
Yeah this package needs to document this assumption, and we need to check that the uses actually work that way.

- [x] gobin
- [ ] java
- [ ] nodejs
- [ ] python
- [ ] ruby

Reply:
just curious why `gobin` is already checked

Reply:
I read through it to see what it did.

Reply:
@RTann was this checked?

Reply:
I think I'd need more clarity on the exact ask. Depending on the image builder, the file may either be updated or deleted and recreated. The changes I made attempts to account for both. Not sure if that's what the question is asking about or something else